src/lib/parsers/Result.ts

```diff
@@ -68,3 +68,29 @@ export function isOk<T, E>(x: Result<T, E>): x is Ok<T> {
 export function isErr<T, E>(x: Result<T, E>): x is Err<E> {
 	return !x.success;
 }
+
+/**
+ * Creates a {@link Result} out of a callback.
+ * @typeparam T The result's type.
+ * @typeparam E The error's type.
+ */
+export function from<T, E = Error>(cb: (...args: unknown[]) => T): Result<T, E> {
+	try {
+		return ok(cb());
+	} catch (error) {
+		return err(error as E);
+	}
+}
+
+/**
+ * Creates a {@link Result} out of a {@link Promise}.
+ * @typeparam T The result's type.
+ * @typeparam E The error's type.
+ */
+export async function fromAsync<T, E = Error>(promise: Promise<T>): Promise<Result<T, E>> {
```

Comment:
For the sake of consistency with `Result.from`, perhaps also support a callback returning `Awaited<T>`?

Reply:
So, something like this?
```ts
export async function fromAsync<T, E = Error>(promiseOrCb: Awaited<T>): Promise<Result<T, E>> {
  if (isFunction(promiseOrCb)) {
    promiseOrCb = promiseOrCb();
  }

  // ...
}
```

Reply:
Assignments aren't needed, you can do this:
```typescript
return ok(await (typeof promiseOrCb === 'function' ? promiseOrCb() : promiseOrCb));
```

Reply:
Alright. Would you rather use `typeof` or the `@sapphire/utilities` function? Also, is `promiseOrCb` a good variable name? It seems kinda verbose to me but idk of a better one

Reply:
They're the same thing, but sure, use `isFunction`.

And the name is fine to me.